drivers/modem/ublox-sara-r4.c

```diff
@@ -1205,13 +1276,97 @@ static bool offload_is_supported(int family, int type, int proto)
 NET_SOCKET_REGISTER(ublox_sara_r4, AF_UNSPEC, offload_is_supported,
 		    offload_socket);
 
+#if defined(CONFIG_DNS_RESOLVER)
+/* TODO: This is a bare-bones implementation of DNS handling
+ * We ignore most of the hints like ai_family, ai_protocol and ai_socktype.
+ * Later, we can add additional handling if it makes sense.
+ */
+static int offload_getaddrinfo(const char *node, const char *service,
+			       const struct addrinfo *hints,
+			       struct addrinfo **res)
+{
+	struct modem_cmd cmd = MODEM_CMD("+UDNSRN: ", on_cmd_dns, 1U, ",");
+	u32_t port = 0U;
+	int ret;
+	/* AT+UDNSRN=#,"[+128]"\r+NUL == 17 + 128 == 145
+	 * (allows for domain name of up to 128 characters)
+	 */
+	char sendbuf[145];
+
+	/* init result */
+	(void)memset(&result, 0, sizeof(result));
+	(void)memset(&result_addr, 0, sizeof(result_addr));
+	/* FIXME: Hard-code DNS to return only IPv4 */
+	result.ai_family = AF_INET;
+	result_addr.sa_family = AF_INET;
+	result.ai_addr = &result_addr;
+	result.ai_addrlen = sizeof(result_addr);
+	result.ai_canonname = result_canonname;
+	result_canonname[0] = '\0';
+
+	/* check to see if node is an IP address */
+	if (net_addr_pton(result.ai_family, node,
+			  &((struct sockaddr_in *)&result_addr)->sin_addr)
+	    == 1) {
+		*res = &result;
+		return 0;
+	}
+
+	/* user flagged node as numeric host, but we failed net_addr_pton */
+	if (hints && hints->ai_flags & AI_NUMERICHOST) {
+		return EAI_NONAME;
+	}
+
+	if (service) {
+		port = strtol(service, NULL, 10);
```

Comment:
Could we call `modem_atoi()` here so that we can handle error cases properly.

Reply:
Yep.  Will change this.